base_layer/wallet_ffi/src/lib.rs

```diff
@@ -860,6 +861,61 @@ pub unsafe extern "C" fn seed_words_create() -> *mut TariSeedWords {
     Box::into_raw(Box::new(TariSeedWords(Vec::new())))
 }
 
+/// Create a TariSeedWords instance containing the entire mnemonic wordlist for the requested language
+///
+/// ## Arguments
+/// `language` - The required language as a string
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `TariSeedWords` - Returns the TariSeedWords instance containing the entire mnemonic wordlist for the requested
+/// language.
+///
+/// # Safety
+/// The ```string_destroy``` method must be called when finished with a string from rust to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn get_mnemonic_wordlist_for_language(
+    language: *const c_char,
+    error_out: *mut c_int,
+) -> *mut TariSeedWords {
+    use tari_key_manager::mnemonic_wordlists;
+
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+
+    let mut seed_words_vec = Vec::new();
+    if language.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("mnemonic wordlist".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+    } else {
+        let language_string = CStr::from_ptr(language).to_str().unwrap();
```

Comment:
is it safe to unwrap this?

Reply:
Nothing is safe in an unsafe method! 

If this fails we need to panic because something is wrong with the underlying memory that is being passed to us so currently throughout the library we make that the clients problem.  We could make these safer though throughout the lib and return from the function gracefully. I will make a ticket to do that throughout though.

Reply:
@philipr-za We can't panic through the FFI, it results in undefined behavior. We always have to gracefully handle an error.